src/ray/core_worker/object_interface.cc

```diff
@@ -129,27 +127,47 @@ Status CoreWorkerObjectInterface::Wait(const std::vector<ObjectID> &ids, int num
     }
   }
 
-  auto current_timeout_ms = timeout_ms;
+  auto wait_from_store_providers = [&ids, &object_ids_per_store_provider, &num_objects,
```

Comment:
Is there a reason why we're using a lambda here instead of just writing a normal function?

Reply:
It's effectively a helper function, I'm feeling it's kind of coupled with the implementation of `CoreWorkerObjectInterface::Wait` thus I use a lambda here instead of creating a new function. I can change it to a member function if you feel strong on this.

Reply:
I see your point, but I feel strongly that we should reserve lambda functions for special cases where we need to pass around functions/callbacks because uses like this hurt readability (people are less familiar with lambdas, documentation won't be as good without header comments etc). It's also easier to accidentally introduce bugs with lambdas due to the somewhat nuanced scoping semantics that people aren't as familiar with as they are the arguments to functions/methods.

Reply:
OK, it's true that this doesn't has to be a lambda, will change to a function instead.

Reply:
updated.